src/main/kotlin/net/atos/zac/flowable/cmmn/AanvullendeInformatieTaskListener.kt

```diff
@@ -0,0 +1,95 @@
+package net.atos.zac.flowable.cmmn
+
+import net.atos.client.zgw.zrc.model.Zaak
+import net.atos.zac.configuratie.ConfiguratieService.Companion.STATUSTYPE_OMSCHRIJVING_AANVULLENDE_INFORMATIE
+import net.atos.zac.configuratie.ConfiguratieService.Companion.STATUSTYPE_OMSCHRIJVING_INTAKE
+import net.atos.zac.flowable.FlowableHelper
+import net.atos.zac.flowable.processengine.ProcessEngineLookupImpl.getCmmnEngineConfiguration
+import nl.lifely.zac.util.NoArgConstructor
+import org.flowable.cmmn.api.runtime.PlanItemInstance
+import org.flowable.engine.delegate.TaskListener
+import org.flowable.task.api.TaskInfo
+import org.flowable.task.service.delegate.BaseTaskListener.EVENTNAME_COMPLETE
+import org.flowable.task.service.delegate.BaseTaskListener.EVENTNAME_CREATE
+import org.flowable.task.service.delegate.DelegateTask
+import java.util.logging.Logger
+
+@NoArgConstructor
+class AanvullendeInformatieTaskListener : TaskListener {
+
+    companion object {
+        private val LOG = Logger.getLogger(AanvullendeInformatieTaskListener::class.java.name)
+
+        private const val STATUS_TOELICHTING = "Status gewijzigd"
+        private const val TASK_AANVULLENDE_INFORMATIE_ID = "AANVULLENDE_INFORMATIE"
+    }
+
+    override fun notify(delegateTask: DelegateTask) {
+        LOG.fine {
+            "AanvullendeInformatie task ${delegateTask.id} in state ${delegateTask.state} was ${delegateTask.eventName}"
+        }
+        (delegateTask as TaskInfo).let { taskInfo ->
```

Comment:
I would suggest using `with` here, so that the context of the block will be the `TaskInfo` object and can be referenced through `this` (or left out), ie:

```kotlin
with(delegateTask as TaskInfo) {
    when (eventName) {
        EVENTNAME_CREATE -> createdEvent(subScopeId)
        EVENTNAME_COMPLETE -> completedEvent(scopeId, subScopeId)
    }
}
```